### PR TITLE
feat(route/chikubi): refactor URL fetching and add new routes

### PR DESCRIPTION
**Pull request by SnowAgar25**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/chikubi
/chikubi/best
/chikubi/video
/chikubi/vr
/chikubi/comic
/chikubi/voice
/chikubi/cg
/chikubi/tag/ドリームチケット
/chikubi/category/nipple-lesbian
/chikubi/nipple-video-category/cat-nipple-video-god
/chikubi/nipple-video-maker/nipple-video-maker-nh
/chikubi/search/ギャップ
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [x] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

### CODE REVIEW

**Comment by SnowAgar25** [comment]
Hi @TonyRL , could you review this PR when you have time? Thank you🙏

File: lib/routes/chikubi/category.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    const url = `/category/${keyword}`;
+
+    const feed = await new Parser().parseURL(`${baseUrl}${url}/feed`);
+
+    const list = feed.items.map((item) => ({
+        title: item.title,
+        link: item.link,
+    }));
+
+    const items = await processItems(list);
```
Comment (TonyRL) [inline lib/routes/chikubi/category.ts:42]: Please use [WordPress REST API](https://developer.wordpress.org/rest-api/) instead.
The API endpoints are available at `/wp-json/wp/v2`.

For categories, you can find the category id first using `/wp-json/wp/v2/categories` with the search parameter `slug`. The id for `nipple-lesbian` is `9`. Then you can list recent posts with the required categories using `/wp-json/wp/v2/posts` with the search parameter `categories`, which will be `9` for `nipple-lesbian`. If you want plain categories and tags for each post in the posts API, you can add the `_embed` search parameter to `/wp-json/wp/v2/posts`

To adjust the no. of articles returned to the posts API, you can pass the `per_page` search parameter to `/wp-json/wp/v2/posts`.

File: lib/routes/chikubi/index.ts [hunk 2]
```diff
@@ -1,13 +1,12 @@
 import { Route, Data } from '@/types';
-import got from '@/utils/got';
-import { load } from 'cheerio';
 import { processItems } from './utils';
+import Parser from 'rss-parser';
```
Comment (TonyRL) [inline lib/routes/chikubi/index.ts:3]: Please use `@/utils/rss-parser` instead.

File: lib/routes/chikubi/nipple-video-category.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import { Route, Data } from '@/types';
+import { processItems } from './utils';
+import Parser from 'rss-parser';
```
Comment (TonyRL) [inline lib/routes/chikubi/nipple-video-category.ts:3]: Please use `@/utils/rss-parser` instead.

File: lib/routes/chikubi/nipple-video-maker.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import { Route, Data } from '@/types';
+import { processItems } from './utils';
+import Parser from 'rss-parser';
```
Comment (TonyRL) [inline lib/routes/chikubi/nipple-video-maker.ts:3]: Please use `@/utils/rss-parser` instead.

File: lib/routes/chikubi/search.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+    const response = await got.get(searchUrl);
+    const searchResults: SearchItem[] = response.data;
+
+    const list = searchResults.map((item) => ({
+        title: item.title,
+        link: item.url,
+    }));
+
+    const items = await processItems(list);
```
Comment (TonyRL) [inline lib/routes/chikubi/search.ts:41]: Please use [WordPress REST API](https://developer.wordpress.org/rest-api/) as well.
In the response of `/search`, it also provides each post's id which you can use `/wp/v2/posts/(?P<id>[\d]+)` to obtain the content

File: lib/routes/chikubi/tag.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    const url = `/tag/${encodeURIComponent(keyword)}`;
+
+    const feed = await new Parser().parseURL(`${baseUrl}${url}/feed`);
+
+    const list = feed.items.map((item) => ({
+        title: item.title,
+        link: item.link,
+    }));
+
+    const items = await processItems(list);
```
Comment (TonyRL) [inline lib/routes/chikubi/tag.ts:42]: Please use [WordPress REST API](https://developer.wordpress.org/rest-api/) instead.

For tags, you can find the tag id first using `/wp-json/wp/v2/tags` with the search parameter `slug`. The id for `ドリームチケット` is `163`. Then you can list recent posts with the required categories using `/wp-json/wp/v2/posts` with the search parameter `categories`, which will be `163` for `ドリームチケット`. If you want plain categories and tags for each post in the posts API, you can add the `_embed` search parameter to `/wp-json/wp/v2/posts`

To adjust the no. of articles returned to the posts API, you can pass the `per_page` search parameter to `/wp-json/wp/v2/posts`.

**Pull request merged**